### PR TITLE
Time, part 4: Switch to moment.js

### LINKED ISSUE #85: Show timezone on activity display
### Issue summary

Show activity timezone on the activity display screen. 

### Expected behavior

An activity's time should be displayed as "Wednesday, May 32nd, 2020 4:56 PM PST".

### Actual behavior

Time displayed as displayed as "Wednesday, May 32nd, 2020 4:56 PM".

This should be a fairly straightforward fix.

### PR DESCRIPTION
**Pull request by ananyeet**
### What is a quick description of the change?

This switches all time.js functions to momentJS utilities for consistency and simplicity.

### Is this fixing an issue?

Fixes #85: Displays timezone abbreviation (CST, PST, EST, etc.) with full time.

### Are there more details that are relevant?

none

### Check lists (check `x` in `[ ]` of list items)

- [x] Test written/updating
- [x] Tests passing
- [x] Coding style (indentation, etc)

Please explain why any are not present, if any.

### CODE REVIEW

File: frontend/src/components/Utils/time.js [hunk 1]
```diff
@@ -10,54 +10,30 @@ import { firestore } from 'firebase';
  * @returns {string} Time formatted into a string like '10:19 AM'.
```
Comment (zghera) [inline frontend/src/components/Utils/time.js:10]: Did you mean to change all of the msTimestamps to `string`s except this one?

File: frontend/src/components/Utils/time.js [hunk 2]
```diff
@@ -88,9 +64,9 @@ export function timezonesForCountry(countryName) {
  */
 export function getDateBarebones(msTimestamp, timezone=null) {
   if (timezone === null) {
-    return moment.tz(parseFloat(msTimestamp), '').format('YYYY-MM-DD');
+    timezone = ''; // Use GMT. 
   }
-  return moment.tz(parseFloat(msTimestamp), timezone).format('YYYY-MM-DD');
+  return moment.tz(parseFloat(msTimestamp), timezone).format(moment.HTML5_FMT.DATE);
```
Comment (zghera) [inline frontend/src/components/Utils/time.js:69]: If these are the same, why did you change it? I think it was more clear the other way. 

If it is different though, you need to update the documentation.

File: frontend/src/components/Utils/time.js [hunk 3]
```diff
@@ -102,9 +78,9 @@ export function getDateBarebones(msTimestamp, timezone=null) {
  */
 export function get24hTime(msTimestamp, timezone=null) {
   if (timezone === null) {
-    return moment.tz(parseFloat(msTimestamp), '').format('HH:mm');
+    timezone = ''; // Use GMT. 
   }
-  return moment.tz(parseFloat(msTimestamp), timezone).format('HH:mm');
+  return moment.tz(parseFloat(msTimestamp), timezone).format(moment.HTML5_FMT.TIME);
```
Comment (zghera) [inline frontend/src/components/Utils/time.js:83]: Same comment as above.

File: frontend/src/components/Utils/time.js [hunk 4]
```diff
@@ -10,54 +10,30 @@ import { firestore } from 'firebase';
...
  */
 export function timestampToTimeFormatted(msTimestamp, timezone = 'America/New_York') {
-  const date = new Date(msTimestamp);
-  const formatOptions = { 
-    hour: 'numeric', 
-    minute: '2-digit', 
-    timeZone: timezone
-  };
-  return date.toLocaleTimeString('en-US', formatOptions);
+  return moment.tz(parseFloat(msTimestamp), timezone).format('LT');
```
Comment (keiffer01) [inline frontend/src/components/Utils/time.js:13]: I think it might help make it clearer to comment what the string in the format function means? Since right now "LT" doesn't mean anything to me.

File: frontend/src/components/Utils/time.js [hunk 5]
```diff
@@ -10,54 +10,30 @@ import { firestore } from 'firebase';
...
-  const date = new Date(msTimestamp);
-  const formatOptions = { 
-    weekday: 'long', 
-    year: 'numeric', 
-    month: 'long', 
-    day: 'numeric',  
-    timeZone: timezone
-  };
-  return date.toLocaleDateString('en-US', formatOptions);
+  return moment.tz(parseFloat(msTimestamp), timezone).format('dddd, LL');
```
Comment (keiffer01) [inline frontend/src/components/Utils/time.js:24]: As above.

File: frontend/src/components/Utils/time.js [hunk 6]
```diff
@@ -10,54 +10,30 @@ import { firestore } from 'firebase';
...
-    weekday: 'long', 
-    year: 'numeric', 
-    month: 'long', 
-    day: 'numeric',  
-    hour: 'numeric', 
-    minute: '2-digit', 
-    timeZone: timezone
-  };
-  return date.toLocaleString('en-US', formatOptions);
+  return moment.tz(parseFloat(msTimestamp), timezone).format('LLLL z');
```
Comment (keiffer01) [inline frontend/src/components/Utils/time.js:36]: As above.

File: frontend/src/components/Utils/time.js (hunk 2 above)
Comment (ananyeet) [reply frontend/src/components/Utils/time.js:69]: Consistency. Plus, if this format changes for whatever reason, we don't have to manually go and update it.

File: frontend/src/components/Utils/time.js [hunk 7]
```diff
@@ -5,59 +5,43 @@ import { firestore } from 'firebase';
...
  */
 export function timestampToTimeFormatted(msTimestamp, timezone = 'America/New_York') {
-  const date = new Date(msTimestamp);
-  const formatOptions = { 
-    hour: 'numeric', 
-    minute: '2-digit', 
-    timeZone: timezone
-  };
-  return date.toLocaleTimeString('en-US', formatOptions);
+  // Formats from https://momentjs.com/docs/#/displaying/format/
```
Comment (Hiram-Silvey) [inline frontend/src/components/Utils/time.js:13]: Add period.

File: frontend/src/components/Utils/time.js [hunk 8]
```diff
@@ -5,59 +5,43 @@ import { firestore } from 'firebase';
...
-  const date = new Date(msTimestamp);
-  const formatOptions = { 
-    weekday: 'long', 
-    year: 'numeric', 
-    month: 'long', 
-    day: 'numeric',  
-    timeZone: timezone
-  };
-  return date.toLocaleDateString('en-US', formatOptions);
+  // Formats from https://momentjs.com/docs/#/displaying/format/
```
Comment (Hiram-Silvey) [inline frontend/src/components/Utils/time.js:26]: Add period.

File: frontend/src/components/Utils/time.js [hunk 9]
```diff
@@ -5,59 +5,43 @@ import { firestore } from 'firebase';
...
 
 /** 
  * Format a timestamp (in milliseconds) into a pretty string.
  * 
- * @param {int} msTimestamp 
+ * @param {number} msTimestamp 
  * @param {string} timezone 
  * @returns {string} Time formatted into a string like 
- * 'Monday, January 19, 1970, 02:48 AM'
+ * 'Monday, January 19, 1970 02:48 AM PST'
```
Comment (Hiram-Silvey) [inline frontend/src/components/Utils/time.js:38]: Add period.

File: frontend/src/components/Utils/time.js [hunk 10]
```diff
@@ -5,59 +5,43 @@ import { firestore } from 'firebase';
...
-    weekday: 'long', 
-    year: 'numeric', 
-    month: 'long', 
-    day: 'numeric',  
-    hour: 'numeric', 
-    minute: '2-digit', 
-    timeZone: timezone
-  };
-  return date.toLocaleString('en-US', formatOptions);
+  // Formats from https://momentjs.com/docs/#/displaying/format/
```
Comment (Hiram-Silvey) [inline frontend/src/components/Utils/time.js:41]: Add period.

File: frontend/src/components/Utils/time.js [hunk 11]
```diff
@@ -4,63 +4,47 @@ import { firestore } from 'firebase';
...
+  // LL = "January 19, 1970"
+  return moment.tz(parseFloat(msTimestamp), timezone).format('dddd, LL');
 }
 
 /** 
  * Format a timestamp (in milliseconds) into a pretty string.
+ * Example: 'Monday, January 19, 1970 02:48 AM PST'.
  * 
- * @param {int} msTimestamp 
+ * @param {number} msTimestamp # propagate from #88
```
Comment (zghera) [inline frontend/src/components/Utils/time.js:39]:
```suggestion
 * @param {number} msTimestamp // propagate from #88
```
